### PR TITLE
Clean up some low level dependencies

### PR DESCRIPTION
**Pull request by james7132**
# Objective
Minimize the number of dependencies low in the tree.

## Solution

 * Remove the dependency on rustc-hash in bevy_ecs (not used) and bevy_macro_utils (only used in one spot).
 * Deduplicate the dependency on `sha1_smol` with the existing blake3 dependency already being used for bevy_asset.
 * Remove the unused `ron` dependency on `bevy_app`
 * Make the `serde` dependency for `bevy_ecs` optional. It's only used for serializing Entity.
 * Change the `wgpu` dependency to `wgpu-types`, and make it optional for `bevy_color`.
 * Remove the unused `thread-local` dependency on `bevy_render`.
 * Make multiple dependencies for `bevy_tasks` optional and enabled only when running with the `multi-threaded` feature. Preferably they'd be disabled all the time on wasm, but I couldn't find a clean way to do this.

---

## Changelog
TODO 

## Migration Guide
TODO

### CODE REVIEW

File: crates/bevy_ecs/src/entity/mod.rs [hunk 1]
```diff
@@ -37,7 +37,9 @@
 //! [`EntityWorldMut::remove`]: crate::world::EntityWorldMut::remove
 mod map_entities;
 #[cfg(feature = "bevy_reflect")]
-use bevy_reflect::{Reflect, ReflectDeserialize, ReflectSerialize};
+use bevy_reflect::Reflect;
+#[cfg(all(feature = "bevy_reflect", feature = "serde"))]
+use bevy_reflect::{ReflectDeserialize, ReflectSerialize};
```
Comment (pablo-lua) [inline crates/bevy_ecs/src/entity/mod.rs:42]: Can we use prelude here? This looks more cleaner IMO
```suggestion
use bevy_reflect::prelude::*;
```

File: crates/bevy_ecs/src/entity/mod.rs [hunk 2]
```diff
@@ -364,6 +368,7 @@ impl From<Entity> for Identifier {
     }
 }
 
+#[cfg(feature = "serde")]
 impl Serialize for Entity {
```
Comment (pablo-lua) [inline crates/bevy_ecs/src/entity/mod.rs:372]: we can have serde::Serialize here and remove the feature import

File: crates/bevy_ecs/src/entity/mod.rs [hunk 3]
```diff
@@ -373,6 +378,7 @@ impl Serialize for Entity {
     }
 }
 
+#[cfg(feature = "serde")]
 impl<'de> Deserialize<'de> for Entity {
```
Comment (pablo-lua) [inline crates/bevy_ecs/src/entity/mod.rs:382]: Same here, this way we will not have feature importing

**Review by pablo-lua — APPROVED** [review]
Just some design questions. Overall, all good.

File: crates/bevy_ecs/src/entity/mod.rs (hunk 1 above)
Comment (james7132) [reply crates/bevy_ecs/src/entity/mod.rs:42]: Wildcard imports, particularly in libraries, are, in my opinion, not something we should be doing. They're not as bad as they are in other languages, but they still have a sizable number of issues in library code that I'm not comfortable with.

See https://rust-lang.github.io/rust-clippy/master/index.html#/wildcard_imports.

File: crates/bevy_ecs/src/entity/mod.rs (hunk 2 above)
Comment (james7132) [reply crates/bevy_ecs/src/entity/mod.rs:372]: They're still required for reflecting the traits.

File: crates/bevy_ecs/src/entity/mod.rs (hunk 3 above)
Comment (james7132) [reply crates/bevy_ecs/src/entity/mod.rs:382]: They're still required for reflecting the traits.

**Review by mockersf — APPROVED** [review]

**Comment by BD103** [comment]
Removing the breaking change label. All of the changes are backwards compatible except for the hash algorithm change in `AnimationTargetId::from_names()` which was added after 0.13 so it's not actually breaking.

**Pull request merged**